nptdms/test/test_example_files.py

```diff
@@ -33,6 +35,47 @@ def test_raw_format():
                                     0.2517777])
 
 
+def tdms_files_assert_equal(tdms1: TdmsFile, tdms2: TdmsFile):
+    """Assert that two TdmsFile instances are equal"""
+    # verify file properties
+    for p in tdms1.properties:
+        np.testing.assert_equal(tdms1.properties[p], tdms2.properties[p])
+    # verify group
+    for group in tdms1.groups():
+        # verify group properties
+        for p in group.properties:
+            np.testing.assert_equal(group.properties[p], tdms2[group.name].properties[p])
+        # verify channels
+        for channel in group.channels():
+            # verify channel data
+            np.testing.assert_equal(channel.data, tdms2[group.name][channel.name].data)
+            # verify channel properties
+            for p in channel.properties:
+                np.testing.assert_equal(channel.properties[p], tdms2[group.name][channel.name].properties[p])
+
+
+@pytest.mark.parametrize("tdms_file", [
+    'raw_timestamps.tdms',  # Test defragmentation of a file with raw timestamps
+    # 'raw1.tdms', # <- cannot defragment this file (ValueError: Channel data must be a 1d array)
```

Comment:
@adamreeve note that trying to defragment this file fails due to it having scalar channel.data

Reply:
Right yes we don't yet support writing or defragmenting files with DaqMxRawData